model/edx.go

```diff
@@ -67,6 +67,29 @@ func (e *EdxLMS) GetOAuthConsumerSecret() string {
 	return e.OAuthConsumerSecret
 }
 
+func (e *EdxLMS) GetTeams() map[string]string {
+	return e.Teams
+}
+
+func (e *EdxLMS) GetLearningGroupPrefixes() []*LearningGroupPrefix {
+	// Instantiate array of pointers to LearningGroupPrefix structs
+	var size int = len(e.PersonalChannels.ChannelList)
+	var prefixes []*LearningGroupPrefix = make([]*LearningGroupPrefix, size)
+
+	// For every key in PersonalChannels.ChannelList in the active configuration
```

Comment:
I'm kind of used to documenting for loops like this. I iteratively comment each part of a loop to say what it's doing. I find it easier to see what's happening than having a large comment at the beginning, then trying to match parts of the comment to parts of code. 

ex. 
```
var fruits = ['banana', 'apple', 'orange']
var bowl = [];

// For every fruit
for (let i = 0; i < fruits.length; i++) {
  // Add the fruit to the bowl
  bowl.push(fruits[i]);
}
```



What do you think @mlippert?

Reply:
I think in general I'd be okay with that, and in the example you show, that style makes sense.

In the code though, you're using a for each loop, not a for loop, so it reads more like:

`var fruits = ['banana', 'apple', 'orange']
var bowl = [];

// For every fruit
for  fruit in fruits {
  // Add the fruit to the bowl
  bowl.push(fruits[i]);
}`

The comment at the top is thus not very helpful, since it reads almost identically to the code.

For me at least, the syntax of the language already tells me _that_ you are putting the fruits in the bowl.  It would be more helpful to have something like this, which tells me _why_ you wanted to put the fruits in the bowl. That's more useful for someone coming after, because maybe GoLang 8 released a new SuperBowl structure that does this better, or maybe there's a refactor to remove the function call this bowl was constructed for.

var fruits = ['banana', 'apple', 'orange']
var bowl = [];

// First, put all the fruits into a bowl, so that we can take advantage of the bowl's 
// efficient internal ordering and nice API
// or, so that we can call a function that expects a bowl
for (let i = 0; i < fruits.length; i++) {
  // Add the fruit to the bowl
  bowl.push(fruits[i]);
}

Reply:
Yeah I see your point about the wording, it does imply a 'for each'. Revised the wording, and pulled comments to the beginning of the loop.

Reply:
@brecriffs So I agree w/ @jaedoucette that comments that say exactly what the following line of code do are not helpful.

If the loop is simple then a comment like "Add every type of fruit to the bowl" or "Add one of each fruit to the bowl" or something that describes what you'll have after the loop seems best. If the actions in the loop are more complicated, then a comment about what the loop iterates over "Examine each fruit for bugs and only add fruit that can be kept cold to the bowl" as the loop comment and inside the loop "Get the info on this fruit, and check all the storage conditions for cold" and then "Add cold resistant fruit to bowl" on some code in the loop or perhaps "Skip non cold resistant fruit" depending on the implementation in the loop.

I wouldn't usually want @jaedoucette 's comment about internal ordering and API either unless there's something really special about them. In other words if I were to be putting a bunch of things in a Set object, I would not comment that I'm using a Set in order to reduce the things to a unique list. I'd expect the developer to already understand that as a property of Sets. I might comment "Get all the types of fruit in the bowl" or some such.

Reply:
@mlippert Your suggestions on this example make sense to me. I also wouldn't want comments about the API or internal ordering unless the structures were something unusual.